repo/blob/s3/s3_storage.go

```diff
@@ -312,6 +312,42 @@ func (s *s3Storage) FlushCaches(ctx context.Context) error {
 	return nil
 }
 
+func isAccessDenied(err error) bool {
+	var e minio.ErrorResponse
+	return errors.As(err, &e) && e.Code == "AccessDenied"
```

Comment:
Isn't there a way of using `errors.Is` here instead of explicitly comparing `e.Code`? Is the expectation that `err` will always be a `minio.ErrorResponse` and won't be wrapped?

Reply:
There is no wrap of `minio.ErrorResponse`, see the [minio code here](https://github.com/minio/minio-go/blob/8be815434d9eec0126e6559ea558ba60b18b8992/api-error-response.go#L120).
`minio.ErrorResponse` is not globally defined as errors, so we cannot use `errors.Is` on `minio.ErrorResponse`. We can translate it and define an global error in Kopia code like `translateError`, but eventually we need to check `e.Code` or `e.StatusCode` somewhere.